neuron/src/lib/Neuron/Zettelkasten/Query/Parser.hs

```diff
@@ -55,27 +55,37 @@ queryFromURILink (URILink linkText uri) =
         _ ->
           throwError $ QueryParseError_UnsupportedHost uri
     _ -> pure $ do
-      -- Initial support for the upcoming short links.
+      -- Modern links:
       -- First, we expect that this is inside <..> (so same link text as link)
       guard angleBracketLink
       -- Then, non-relevant parts of the URI should be empty
-      guard
-        `mapM_` [ URI.uriAuthority uri == Left False,
-                  URI.uriFragment uri == Nothing
-                ]
+      guard $ URI.uriFragment uri == Nothing
       let mconn =
             if hasQueryFlag [queryKey|cf|] uri
               then Just OrdinaryConnection
               else Nothing
+          -- Found "z:" without a trailing slash
+          noSlash = URI.uriAuthority uri == Left False
+          -- Found "z:/" instead of "z:"
+          hasSlash = URI.uriAuthority uri == Left True
```

Comment:
Maybe we can use [`isPathAbsolute`](https://hackage.haskell.org/package/modern-uri-0.3.2.0/docs/src/Text.URI.Types.html#isPathAbsolute).

Reply:
That includes `z://foo` which is not supported by the current spec.